src/ray/pubsub/subscriber.cc

```diff
@@ -173,6 +178,16 @@ void Subscriber::MakeLongPollingPubsubConnection(const rpc::Address &publisher_a
   rpc::PubsubLongPollingRequest long_polling_request;
   long_polling_request.mutable_subscriber_address()->CopyFrom(subscriber_address);
 
+  // Update the command in the FIFO order.
+  int64_t updated_commands = 0;
+  while (!commands_.empty() && updated_commands < command_max_batch_size_) {
+    auto &command = commands_.front();
+    auto *new_command = long_polling_request.add_commands();
+    new_command->Swap(command.get());
+    commands_.pop();
+    updated_commands += 1;
+  }
```

Comment:
I have some questions here. What do we do for the rest of the commands_? Do you mean that we'll keep calling this function until there is nothing left?

Reply:
They will be delivered in the next batch. But actually, that's a good point. We should reply right away if there are commands left. I will fix that

Reply:
To be clear, here's the workflow; 

Subscriber batches commands in a long polling "request" -> publisher replies right away although there are no pub messages -> in this way, the subscriber keeps sending long polling "request" until all commands are served. 

I can probably add this comment here.

Reply:
Actually, this will be handled in the next PR where we actually migrate all subscribe to command based subscription.